packages/website/ts/components/fill_order.tsx

```diff
@@ -627,20 +621,21 @@ export class FillOrder extends React.Component<FillOrderProps, FillOrderState> {
 
         const takerTokenAmount = new BigNumber(parsedOrder.signedOrder.takerTokenAmount);
 
-        const signedOrder = this.props.blockchain.portalOrderToSignedOrder(
-            parsedOrder.signedOrder.maker,
-            parsedOrder.signedOrder.taker,
-            parsedOrder.signedOrder.makerTokenAddress,
-            parsedOrder.signedOrder.takerTokenAddress,
-            new BigNumber(parsedOrder.signedOrder.makerTokenAmount),
-            takerTokenAmount,
-            new BigNumber(parsedOrder.signedOrder.makerFee),
-            new BigNumber(parsedOrder.signedOrder.takerFee),
-            new BigNumber(this.state.parsedOrder.signedOrder.expirationUnixTimestampSec),
-            parsedOrder.signedOrder.feeRecipient,
-            parsedOrder.signedOrder.ecSignature,
-            new BigNumber(parsedOrder.signedOrder.salt),
-        );
+        const signedOrder = {
+            exchangeContractAddress: this.props.blockchain.getExchangeContractAddressIfExists(),
+            maker: parsedOrder.signedOrder.maker,
+            taker: parsedOrder.signedOrder.taker,
+            makerTokenAddress: parsedOrder.signedOrder.makerTokenAddress,
+            takerTokenAddress: parsedOrder.signedOrder.takerTokenAddress,
+            makerTokenAmount: new BigNumber(parsedOrder.signedOrder.makerTokenAmount),
+            takerTokenAmount: new BigNumber(parsedOrder.signedOrder.takerTokenAmount),
+            makerFee: new BigNumber(parsedOrder.signedOrder.makerFee),
+            takerFee: new BigNumber(parsedOrder.signedOrder.takerFee),
+            expirationUnixTimestampSec: new BigNumber(this.state.parsedOrder.signedOrder.expirationUnixTimestampSec),
+            feeRecipient: parsedOrder.signedOrder.feeRecipient,
+            ecSignature: parsedOrder.signedOrder.ecSignature,
+            salt: new BigNumber(parsedOrder.signedOrder.salt),
+        };
```

Comment:
This seems duplicitous. Let's just have a helper method that takes a stringified `signedOrder` and returns a `bigNumbered` `signedOrder`.